ui/src/app/utils/runtime-utils.tsx

```diff
@@ -302,14 +355,19 @@ export const useRuntimeStatus = (currentWorkspaceNamespace, currentGoogleProject
   RuntimeStatus | undefined, (statusRequest: RuntimeStatusRequest) => Promise<void>]  => {
   const [runtimeStatus, setRuntimeStatus] = useState<RuntimeStatusRequest>();
   const {runtime} = useStore(runtimeStore);
-
+  const enablePD = serverConfigStore.get().config.enablePersistentDisk;
   // Ensure that a runtime gets initialized, if it hasn't already been.
   useRuntime(currentWorkspaceNamespace);
+  if (enablePD) {
```

Comment:
I believe it is problematic to conditionally install hooks like this. I thought I made this comment alreayd, but I can't find it. Please move the `enablePD` check into the `useDisk` hook instead, inside the use effect too.